pkg/k8s/client.go

```diff
@@ -564,3 +567,42 @@ func (c *Client) GetUpstreamTrafficSetting(namespace *types.NamespacedName) *pol
 	}
 	return nil
 }
+
+// GetMeshRootCertificate returns a MeshRootCertificate resource with namespaced name
```

Comment:
nit: consider adding the k8s client stuff in a separate PR to reduce the size of this one.

Reply:
Created #5127